finality-aleph/src/lib.rs

```diff
@@ -142,6 +143,35 @@ impl aleph_bft::SpawnHandle for SpawnHandle {
     }
 }
 
+#[derive(PartialEq, Eq, Clone, Debug, Hash)]
+pub(crate) struct HashNum<H, N> {
+    pub(crate) hash: H,
+    pub(crate) num: N,
+}
+
+impl<H, N> HashNum<H, N> {
+    pub(crate) fn new(hash: H, num: N) -> Self {
+        HashNum { hash, num }
+    }
+}
+
+pub(crate) type BlockHashNum<B> = HashNum<<B as Block>::Hash, NumberFor<B>>;
```

Comment:
If `HashNum` took `B: Block` as an argument it couldn't have `Hash` derived for it, since `Block` does not imply `Hash`. I think this is also actually cleaner, since `HashNum` as implemented does not actually depend on its components being associated types to a specific `Block` type.

Reply:
Yeah, that's a workaround to this problem :). What's annoying with it is that not only you need to implement `Hash` but also `PartialEq` (clippy :/). 
@pmikolajczyk41 there is a discussion on #dev-hotline about that (or actually me asking for help)